portfolio/src/main/webapp/script.js

```diff
@@ -1,28 +1,32 @@
+
+//this function generates a random quote to display, also making sure that the previously displayed random image is not the same as the current random image.
 function randomizeQuote() {
 
-  var imageIndex = Math.floor(Math.random() * 4) + 1;
+  //ImageNumber has the Image index of the quotes. This generates a number between [1-4]
+  var imageNumber = Math.floor(Math.random() * 4) + 1;
 
-  //To check if an image exists already
+  //To check if an image exists already. The length of the list returned must have some element in it. This means that this is not the first time the button was clicked after page reload. 
   if($('#generated-quote-id').length>0)
   {
-    //Image exists
-    var currentImageIndex = $('#generated-quote-id').attr('src').substr(13,1)
-    while(imageIndex+"" === currentImageIndex)
-    {
-      imageIndex = Math.floor(Math.random() * 4) + 1;
+    var displayedImageIndex = $('#generated-quote-id').attr('src').substr(13,1);    //This gets the number(index) of the image that was displayed right before the button was clicked. 
+
+    while(imageNumber+"" === displayedImageIndex){        //checks if the newly generated index is the same as the previously displayed image. as long as it is the same, it generates a different index
+      imageNumber = Math.floor(Math.random() * 4) + 1;        
```

Comment:
In theory this could loop for a very long time (depending on the implementation Math.random). Is there a better way we could handle this?

This does not need fixed before merging this pull request. This can be done in a separate pull request if you have time.

Reply:
One idea is we can exclude the displayed image number by getting a random number from 1-3.

Reply:
What if ImageIndex is 2, and the displayed ImageIndex is 3? That way, how will you exclude 3 the previous time?